fix(30216): Add more docs for must version of functions (#1698)
helm/helm-www: content/en/docs/chart_template_guide/function_list.md

(mustRegexReplaceAllLiteral)](#regexreplaceallliteral-mustregexreplaceallliteral),
[regexSplit (mustRegexSplit)](#regexsplit-mustregexsplit).

> ### regexMatch

Reviewer: There are links above and this change breaks that. I don't think you should change the headers or add the `### mustRegexMatch` header here.

Returns `true` if the input string contains any match of the regular expression.

```
{{- if regexMatch "^abc$" "abc" -}}
valid: true
{{- end -}}
```
The above produces `valid: true`.

> `regexMatch` will ignore regex compilation errors, and simply return `false` ([ref](https://github.com/Masterminds/sprig/blob/master/regex.go#L7))
> 
> ```
> {{- if regexMatch "[" "abc" -}}
> valid: true
> {{- end -}}
> ```
> The above produces nothing.
> 
> ### mustRegexMatch
> 
> Similar to `regexMatch`, but will return an error to the templating engine if the regex fails to compile:

Reviewer: ```suggestion
In the event of regex compilation errors, `regexMatch` will simply return `false`.
Use `mustRegexMatch` to fail in the event of regex compilation errors.
```

`regexFindAll` panics if there is a problem and `mustRegexFindAll` returns an
error to the template engine if there is a problem.

> ### regexFind

Reviewer: ```suggestion
### regexFind, mustRegexFind
```

The above produces `d1`

> `regexFind` will result in a panic if the regex fails to compile ([ref](https://github.com/Masterminds/sprig/blob/master/regex.go#L30)), whereas
> `mustRegexFind` will return an error to the templating engine if the regex fails to compile ([ref](https://github.com/Masterminds/sprig/blob/master/regex.go#L34)).

Reviewer: I think these references are going to be brittle
```suggestion
The `regexFind` function will result in a panic if the regex fails to compile, whereas
`mustRegexFind` will return an error to the templating engine if the regex fails to compile.
```

> Note that neither function will panic/return an error if the input string does not match the regex. Both functions return an empty string in this case:

Reviewer: This project doesn't strictly use semantic line breaks, but I think it helps.
```suggestion
Note that neither function will panic/return an error if the input string does not match the regex.
Both functions return an empty string in this case:
```
